sdk/client-core/core/src/main/java/io/clientcore/core/util/ServerSentEventUtils.java

```diff
@@ -20,114 +21,123 @@
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Objects;
-import java.util.function.Consumer;
 import java.util.regex.Pattern;
 
 /**
- * Utility class for Server Sent Event handling
+ * Utility class for Server Sent Event handling.
  */
-public final class ServerSentEventUtil {
+public final class ServerSentEventUtils {
     private static final String DEFAULT_EVENT = "message";
     private static final Pattern DIGITS_ONLY = Pattern.compile("^[\\d]*$");
     private static final String LAST_EVENT_ID = "Last-Event-Id";
     public static final String NO_LISTENER_ERROR_MESSAGE = "No ServerSentEventListener attached to HttpRequest to "
-            + "handle the text/event-stream response";
+        + "handle the text/event-stream response";
 
-    private ServerSentEventUtil() {
+    private ServerSentEventUtils() {
     }
 
     /**
-     * Checks if the content type is a text event stream
+     * Checks if the {@code Content-Type} is a text event stream.
      *
-     * @param contentType The content type
-     * @return True if the content type is a text event stream
+     * @param contentType The content type.
+     *
+     * @return {@code true} if the content type is a text event stream.
      */
     public static boolean isTextEventStreamContentType(String contentType) {
         if (contentType != null) {
-            return ContentType.TEXT_EVENT_STREAM.regionMatches(true, 0, contentType, 0, ContentType.TEXT_EVENT_STREAM.length());
+            return ContentType.TEXT_EVENT_STREAM.regionMatches(true, 0, contentType, 0,
+                ContentType.TEXT_EVENT_STREAM.length());
         } else {
             return false;
         }
     }
 
     /**
-     * Processes the text event stream
+     * Processes the text event stream.
      *
-     * @param httpRequest The HTTP Request
-     * @param httpRequestConsumer The HTTP Request consumer
-     * @param inputStream The input stream
-     * @param listener The listener object attached with the httpRequest
-     * @param logger The logger object
+     * @param httpRequest The {@link HttpRequest} to send.
+     * @param httpClient The {@link HttpClient} to send the {@code httpRequest} through.
+     * @param inputStream The {@link InputStream} to read data from.
+     * @param listener The {@link ServerSentEventListener event listener} attached to the {@link HttpRequest}.
+     * @param logger The {@link ClientLogger} object to log errors with.
      */
-    public static void processTextEventStream(HttpRequest httpRequest, Consumer<HttpRequest> httpRequestConsumer,
-        InputStream inputStream, ServerSentEventListener listener, ClientLogger logger) {
-        RetrySSEResult retrySSEResult;
+    public static void processTextEventStream(HttpRequest httpRequest, HttpClient httpClient,
+                                              InputStream inputStream, ServerSentEventListener listener,
+                                              ClientLogger logger) {
```

Comment:
In terms of parameter precedence, I would put `HttpClient` before `HttpRequest`. I also wonder if we are doing the right thing passing in `ClientLogger` here, seeing as it is presumably available via the `HttpRequest` / `RequestOptions`.